src/Ser/SerYaml.jl

```diff
@@ -0,0 +1,293 @@
+module SerYaml
+
+export to_yaml
+
+using Dates
+
+const YAML_NULL = "null"
+const INDENT = "  "
+const INDENT_TYPES = [Pair, AbstractDict, AbstractVector, Tuple, NamedTuple, AbstractSet]
+
+# escaped string
+
+const NEEDESCAPE =
+    Set{UInt8}(UInt8['"', '\\', '\b', '\f', '\n', '\r', '\t'])
+
+function escape_char(b)
```

Comment:
We should consider refactoring this out 🤔

Reply:
makes sense, but I think it can be done as part of a separate issue

Reply:
Rewriting these pieces in a more straightforward manner would have been a boon